### PR TITLE
Adding OTA agent demo to the build system

### PR DESCRIPTION
**Pull request by manvensh**
Issue #, if available:

Description of changes:

1. Added the OTA agent demo to the build system.
2. Updated the OTA agent demo according to the MQTT streams library changes.

Testing
Tested the changes by running both OTA agent and non-agent demo.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kstribrnAmzn** [comment]
Please update the README to add instructions for running the OTA Agent demo.

**Comment by manvensh** [comment]
> Please update the README to add instructions for running the OTA Agent demo.

Updated the README.

File: README.md [hunk 1]
```diff
@@ -9,10 +9,17 @@ allow for customization for new fields and extensibility to new services. Using
...
-coreMQTT. The example will check IoT Core for an existing OTA Job, download its
-associated file, output the file over the command line, and report success back
-to IoT Core.
+This repository contains the following two example OTA orchestrators. Both
+orchestrators use FreeRTOS, coreMQTT and IoT Jobs library.
+
+1. **Simple OTA Orchestrator**: It is a simple orchestrator which checks IoT Core
+for an existing OTA Job, download its associated file, output the file over the
+command line, and report success back to IoT Core.
+2. **OTA Agent Orchestrator**: The OTA agent orchestrator operates by managing
```
Comment (kstribrnAmzn) [inline README.md:18]: I'd mention that this example is designed to mimic the OTA agent found in the old OTA repository

**Review by kstribrnAmzn — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (manvensh) [reply README.md]: Added this in the README.

**Pull request merged**